### PR TITLE
make: Allow building select images

### LINKED ISSUE #1803: make build should be able to build only 1 storage provider
Is this a bug report or feature request?
* Feature Request

**Feature Request**

Are there any similar features already existing:

What should the feature do:  When a developer wants to build, they should be able to only build the container image for a single storage provider, such as Ceph.  Something like `make build ceph` should only build the Ceph image and skip others such as CockroachDB or Minio.  Other potential commands could be scoped to single providers, such as `minikube.sh update ceph`.

What would be solved through this feature: Developers time during iteration could be saved by not having to wait on all containers to build.  This will only get worse as more providers are integrated into Rook.

Does this have an impact on existing features: It should not, `make build` should still behave exactly as it does today (build container images for all providers)

### PR DESCRIPTION
**Pull request by BlaineEXE**
Raise the scope of the `IMAGES` config variable in the
`build/release/Makefile` to the `build/makelib/common.mk` file, and use
the `IMAGES` variable in the `images/Makefile` to allow users to build
only select images.

Examples of new use:
  make IMAGES='ceph ceph-toolbox' build
  make -C build/release IMAGES='ceph ceph-toolbox' publish

The 'images/image.mk' clean.build target was changed to use
`CLEAN_IMAGES` instead of `IMAGES`, as it was conflicting with use of
the parameter in image builds.

Added IMAGES option to top-level make's help text. The text for this
option exceeded 80 lines in the `Makefile` itself, so the help text was
refactored to eliminate `    @echo '<text>'`. The `Makefile` text should
now match exactly what is output to the user including line length.

Added IMAGES option to `INSTALL.md` file's "Build" section and also
moved some text around and added a few extra points of clarification
there to hopefully improve readability, especially for first-time
readers.

Many of the `rook/images/<image>/` directory Makefiles defined a
variable called `IMAGES`, which was unused and causing some developer
confusion with the feature adding the ability to specify which backend
images to build. Remove these unused instances.

Signed-off-by: Blaine Gardner <blaine.gardner@suse.com>

**Which issue is resolved by this Pull Request:**
Resolves #1803

**Checklist:**
- [x] Documentation has been updated, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.
- [ ] `make vendor` does not cause changes.

### CODE REVIEW

File: images/Makefile [hunk 1]
```diff
@@ -38,7 +40,7 @@ cockroachdb.%:
 minio.%:
 	@$(MAKE) -C minio PLATFORM=$*
 
-do.build.images.%: ceph.% ceph-toolbox.% cockroachdb.% minio.% ;
+do.build.images.%: $(foreach i,$(IMAGES), $(i).%);
```
Comment (travisn) [inline images/Makefile]: jenkins seems to be hitting an error during cleanup
```
=== cleaning images for build-88c6720f
Makefile:43: *** target pattern contains no '%'.  Stop.
image.mk:100: recipe for target 'clean.build' failed
make[1]: *** [clean.build] Error 2
Makefile:151: recipe for target 'clean' failed
make: *** [clean] Error 2
```

Comment (BlaineEXE) [reply images/Makefile]: Hmm. I'll look into it.

**Comment by jbw976** [comment]
good stuff @BlaineEXE, thanks for this effort!  @travisn was just complaining about it this morning again haha. Looks like this is addressing #1803. I've added that to the PR description, let me know if I've misunderstood.

**Comment by BlaineEXE** [comment]
I notice that there is also an `IMAGES` variable (exactly like I created) in `build/release/Makefile`. It seems to me that the `build/release/Makefile` is quite separate from the targets of the top-level Makefile, so I think this may still be desired. But I can see value in changing it to be DRY. To do this, I think I would put `IMAGES ?= ...` in the `images/image.mk` file and have `build/release/Makefile` include it. Let me know if that is more appealing.

**Comment by BlaineEXE** [comment]
@jbw976, I didn't see the issue you mentioned. I believe that this is indeed resolving #1803. Thanks for the cross-reference. Also, please feel free to chime in on the double-IMAGES concern I posted just above.

**Comment by travisn** [comment]
@BlaineEXE if renaming the conflicting var to `CLEAN_IMAGES` fixes the issue, it seems sufficient to me. @bassam may have more insight though.

**Comment by BlaineEXE** [comment]
@travisn and @jbw976, I made the changes we talked about this morning. Let me know if it looks good or you see anything else.

File: images/Makefile [hunk 2]
```diff
@@ -16,6 +16,8 @@ include image.mk
 
 PLATFORMS ?= linux_amd64 linux_arm64
 
+IMAGES ?= ceph ceph-toolbox cockroachdb minio
```
Comment (travisn) [inline images/Makefile]: so we can remove this now since image.mk includes common.mk, where this variable is defined?

Comment (BlaineEXE) [reply images/Makefile]: Gah! I guess this got missed in a rebase.

File: build/makelib/common.mk [hunk 3]
```diff
@@ -94,6 +94,9 @@ ifeq ($(origin BUILD_REGISTRY), undefined)
 BUILD_REGISTRY := build-$(shell echo $(HOSTNAME)-$(ROOT_DIR) | shasum -a 256 | cut -c1-8)
 endif
 
+# Select which images (backends) to make; default to all possible images
+IMAGES ?= ceph ceph-toolbox cockroachdb minio
```
Comment (bassam) [inline build/makelib/common.mk]: does this need to go in common.mk? I think it belongs in images/Makefile

**Comment by BlaineEXE** [comment]
I originally had it defined there, but the same line was also defined in
build/release, so I moved the line to common.mkso there wouldn't be
duplication.

**Pull request merged**